### ISSUE TITLE
Kamon Logback logs empty values for context entries that are not in the current context

### ISSUE THREAD

**Issue by yiminyuan** [kamon-io/Kamon#918]
- Scala 2.12.12
- Kamon 2.1.9

We currently use a [JSON logger](https://github.com/logstash/logstash-logback-encoder). If I add this config: `kamon.instrumentation.logback.mdc.copy.entries = [ foo ]`, then `foo` would be logged as an empty value even though `foo` is not in the current context:
```
{
  "foo" : ""
}
```

So the workaround is adding the following code to `logback.xml` and remove `kamon.instrumentation.logback.mdc.copy.entries = [ foo ]`:
```
<conversionRule conversionWord="contextEntry" converterClass="kamon.instrumentation.logback.tools.ContextEntryConverter" />

<pattern>
  <omitEmptyFields>true</omitEmptyFields>
  <pattern>
    {
      "foo":"%contextEntry{foo}"
    }
  </pattern>
</pattern>
```

**Temporary fix**
In https://github.com/kamon-io/Kamon/blob/master/instrumentation/kamon-logback/src/main/scala/kamon/instrumentation/logback/LogbackInstrumentation.scala, it seems only context tags that are in the current context are reported.
```scala
currentContext.tags.iterator().foreach(t => {
  MDC.put(t.key, Tag.unwrapValue(t).toString)
})
```
We have a very simple use case that all K-V pairs have the string type. So we use context tags instead of context entries.

**Expected behavior**
If a context entry is not in the current context, then it should not be logged.

**Proposed fix**
In https://github.com/kamon-io/Kamon/blob/master/instrumentation/kamon-logback/src/main/scala/kamon/instrumentation/logback/LogbackInstrumentation.scala, because of `Context.key[Any](key, "")`, which means `""` is returned by default if `key` is not in the current context. Hence, we can add extra if-check like this to eliminate the issue:
```scala
settings.mdcCopyKeys.foreach { key =>
  currentContext.get(Context.key[Any](key, "")) match {
    case Some(value) =>
      val s = value.toString
      if(s.nonEmpty)
        MDC.put(key, s)
    case keyValue if keyValue != null =>
      val s = keyValue.toString
      if(s.nonEmpty)
        MDC.put(key, s)
    case _ => // Just ignore the nulls.
  }
}
```

**Comment by SimunKaracic** [comment]
Fixed in 2.1.13

**Comment by yiminyuan** [comment]
Awesome! This commit https://github.com/kamon-io/Kamon/commit/85717aa0836c527e784a08cf7dbc46cb131341c2 looks good. Thank you for fixing it!
